includes/class-sensei-quiz.php

```diff
@@ -954,6 +954,20 @@ public function set_user_grades( $quiz_grades, $lesson_id, $user_id = 0 ) {
 
 	}
 
+	/**
+	 * Check if the user has completed the quiz.
+	 *
+	 * @since 4.3.0
+	 *
+	 * @param int|null $quiz_id (Optional) The quiz post ID. Defaults to the current post ID.
+	 * @param int|null $user_id (Optional) The user ID. Defaults to the current user ID.
+	 *
+	 * @return bool
+	 */
+	public function is_completed( int $quiz_id = 0, int $user_id = 0 ): bool {
+		return self::is_quiz_completed( $quiz_id, $user_id );
+	}
```

Comment:
I like the idea of moving away from static and `Sensei()->quiz->is_completed()` reads nicely.

I'm not sure if we are going to do this kind of refactoring, should we deprecate the old method? I don't see a reason to have both. Also, if we are going to deprecate the method, the codebase should be updated to use the new method, and it might be a good idea to make it in a separate PR.

@Automattic/nexus Do you see any potential issues with the above?

Reply:
@donnapep Do you think we should do small code refactorings like this in the future or try to avoid it until we decide how we should move about it?

Reply:
Small refactoring sounds fine, but if there is some new approach being introduced in this PR that we think we should adopt going forward, that should ideally be discussed on a P2 so that everyone has a chance to chime in. We should try to avoid making important decisions on GitHub. Not sure if you would consider this change to fall into that category or not.

Reply:
@donnapep Sounds good to have a discussion first. 👍 

@gabrielcaires Would you like to initiate the discussion? Also, I guess you could revert the refactoring changes until we decide on how we should go about it and make a separate PR afterward?

Reply:
Ok, I am just a bit confused because I think I am not using a new pattern, on the same file we have other methods using the same strategy:
https://github.com/Automattic/sensei/blob/c88940faa316a7bae46f4ebe43d4a85c72148e27/includes/class-sensei-quiz.php#L1047-L1047

On the file that I using my method,  there are both kind of calls (static and by instance)
https://github.com/Automattic/sensei/blob/c88940faa316a7bae46f4ebe43d4a85c72148e27/includes/class-sensei-question.php#L1140-L1155.

Reply:
Your case is a bit different I think, you're introducing a new method that is replacing an old one. For me, this creates the following questions:

- Is this better for developers?
- Are we going to deprecate the old method to make it more obvious what we will support in the future?

---

```php
$all_user_grades = self::get_user_grades( $lesson_id, $user_id ); 
```

The `get_user_grades` method is non-static. Calling a non-static method with double colon is actually deprecated in PHP 7.0 and removed in PHP 8.0. We should update that I guess...

Reply:
@m1r0 PR updated. Can you check again?